src/lib.rs

```diff
@@ -481,110 +484,149 @@ define_sentence_type_enum!(
     ///                      VTG | WCV | WNC | WPL | XDR | XTE | XTR |
     /// Wind: MWV | VPW | VWR |
     /// Date and Time: GDT | ZDA | ZFO | ZTG |
-    enum SentenceType {
-        AAM,
-        ABK,
-        ACA,
-        ACK,
-        ACS,
-        AIR,
-        ALM,
-        ALR,
-        APA,
-        APB,
-        ASD,
-        BEC,
-        BOD,
-        BWC,
-        BWR,
-        BWW,
-        CUR,
-        DBK,
-        DBS,
-        DBT,
-        DCN,
-        DPT,
-        DSC,
-        DSE,
-        DSI,
-        DSR,
-        DTM,
-        FSI,
-        GBS,
-        GGA,
-        GLC,
-        GLL,
-        GMP,
-        GNS,
-        GRS,
-        GSA,
-        GST,
-        GSV,
-        GTD,
-        GXA,
-        HDG,
-        HDM,
-        HDT,
-        HMR,
-        HMS,
-        HSC,
-        HTC,
-        HTD,
-        LCD,
-        LRF,
-        LRI,
-        LR1,
-        LR2,
-        LR3,
-        MLA,
-        MSK,
-        MSS,
-        MWD,
-        MTW,
-        MWV,
-        OLN,
-        OSD,
-        ROO,
-        RMA,
-        RMB,
-        RMC,
-        ROT,
-        RPM,
-        RSA,
-        RSD,
-        RTE,
-        SFI,
-        SSD,
-        STN,
-        TLB,
-        TLL,
-        TRF,
-        TTM,
-        TUT,
-        TXT,
-        VBW,
-        VDM,
-        VDO,
-        VDR,
-        VHW,
-        VLW,
-        VPW,
-        VSD,
-        VTG,
-        VWR,
-        WCV,
-        WNC,
-        WPL,
-        XDR,
-        XTE,
-        XTR,
-        ZDA,
-        ZDL,
-        ZFO,
-        ZTG,
-    }
+    AAM, 1;
```

Comment:
May be it is possible to calculate bits automatically (to reduce typo probability)?

Something like:
```rust
macro_rules! count_idents {
    () => (1);
   ($head:ident) => (2);
    ($head:ident $(, $tail:ident)*) => (1 + count_ident!($($tail),*));
}

macro_rules! define_enum {
    () => have_no_idea;
    ($head:ident $(, $tail:ident)*) => ($head = count_idents!($($tail),*), call_recursively));
}
```

and it would be also nice to have static assert about max enum <= 128

Reply:
I fixed it with another approach from the Little Rust Macro Book as your's would cause recursion issues actually....the solution is a little hacky though, I'd be open for suggestion on where to place the macro calls.